src/m365/spo/commands/site/site-recyclebinitem-restore.ts

```diff
@@ -50,11 +58,19 @@ class SpoSiteRecycleBinItemRestoreCommand extends SpoCommand {
           return isValidSharePointUrl;
         }
 
-        const invalidIds = formatting
-          .splitAndTrim(args.options.ids)
-          .filter(id => !validation.isValidGuid(id));
-        if (invalidIds.length > 0) {
-          return `The following IDs are invalid: ${invalidIds.join(', ')}`;
+        if (args.options.ids) {
+          const invalidIds = formatting
+            .splitAndTrim(args.options.ids)
+            .filter(id => !validation.isValidGuid(id));
+
+          if (invalidIds.length > 0) {
+            return `The following IDs are invalid: ${invalidIds.join(', ')}`;
+          }
+        }
+
+        if ((!args.options.ids && !args.options.allPrimary && !args.options.allSecondary)
+          || (args.options.ids && (args.options.allPrimary || args.options.allSecondary))) {
+          return 'Specify ids or allPrimary and/or allSecondary';
```

Comment:
Maybe let's use a more clear message: `Option 'ids' cannot be used with 'allPrimary' or 'allSecondary'.`